plugins/secrets/index.js

```diff
@@ -9,8 +11,59 @@ const createRoute = require('./create');
  * @param  {Function} next              Function to call when done
  */
 exports.register = (server, options, next) => {
+    /**
+     * Check if a credential has access to a secret
+     * @method canAccess
+     * @param {Object} credentials              Credential object from Hapi
+     * @param {String} credentials.username     Username of the person logged in (or build ID)
+     * @param {Array}  credentials.scope        Scope of the credential (user, build, admin)
+     * @param {String} [credentials.pipelineId] If credential is a build, this is the pipeline ID
+     * @param {String} [credentials.jobId]      If credential is a build, this is the job ID
+     * @param {String} [credentials.isPR]       If credential is a build, this is true if a PR
+     */
+    server.expose('canAccess', (credentials, secret) => {
+        const userFactory = server.root.app.userFactory;
+        const pipelineFactory = server.root.app.pipelineFactory;
+        const username = credentials.username;
+        const scope = credentials.scope;
+
+        return pipelineFactory.get(secret.pipelineId).then(pipeline => {
+            if (!pipeline) {
+                throw boom.notFound(`Pipeline ${secret.pipelineId} does not exist`);
+            }
+
+            if (scope.includes('user')) {
+                return userFactory.get({ username }).then(user => {
+                    if (!user) {
+                        throw boom.notFound(`User ${username} does not exist`);
+                    }
+
+                    return user.getPermissions(pipeline.scmUrl).then(permissions => {
+                        if (!permissions.push) {
+                            throw boom.forbidden(`User ${username} does not have `
+                                + 'write access to this repository');
+                        }
+
+                        return false;
```

Comment:
What `canAccess` actually returns is whether to show the secret value depends on the scope. However just by reading the function name and description, I will think it returns whether the credential can access the secret. Can we document this more clearly?